DEVEL.rst

```diff
@@ -8,132 +8,176 @@ If something does not work as expected, please create an issue or
+   $ . devel/bin/activate
+
 
 Continue executing the following commands in the virtualenv you just created.
 
 
-## Install Commissaire
-```
-cd commissaire
-pip install -e .
-popd
-```
+Install Commissaire
+===================
 
+.. code-block::
 
-### Install Commissaire Service
-```
-cd commissaire-service
-pip install -e .
-tools/etcd_init.sh
-```
+   $ cd commissaire
+   $ pip install -e .
+   $ popd
 
-Edit your storage configuration to point to your etcd instance
-```
-cp conf/storage.conf mystorage.conf
-```
-Note: point server_url to http://127.0.0.1:2379 (not https)
+
+Install Commissaire Service
+---------------------------
+
+.. code-block::
+
+   $ cd commissaire-service
+   $ pip install -e .
+   $ tools/etcd_init.sh
+
+
+Edit your storage configuration to point to your etcd instance:
+
+.. code-block::
+
+   $ cp conf/storage.conf mystorage.conf
+
+
+**Note**: Point ``server_url`` to ``http://127.0.0.1:2379`` (not https)
 
 
 Start the service
-```
-commissaire-storage-service -c mystorage.conf &
-popd
-```
 
-### Install Commissaire Server
-```
-cd commissaire-http
-pip install -e .  # Install commissaire-http into the virtualenv
-```
+.. code-block::
+
+   $ commissaire-storage-service -c mystorage.conf &
+   $ popd
+
+
+Install Commissaire Server
+--------------------------
+
+.. code-block::
+
+   $ cd commissaire-http
+   $ pip install -e .  # Install commissaire-http into the virtualenv
+
 
 Edit the configuration to point to your redis instance
-```
-cp conf/commissaire.conf config.conf
-```
-Note: if locally installed you do not need to change anything
 
-Start the service
-```
-commissaire-server -c config.conf &
-popd
-```
+.. code-block::
+
+   $ cp conf/commissaire.conf config.conf
+
+**Note:** If locally installed you do not need to change anything
+
+Start the service:
+
+.. code-block::
+
+   $ commissaire-server -c config.conf &
+   $ popd
+
+
+Run testcases for Commissaire Server
+++++++++++++++++++++++++++++++++++++
+
 
-#### Run testcases for Commissaire Server
-Note that you can use `tox` to run testcases for this project.
+**Note**: that you can use ``tox`` to run testcases for this project.
 
-Install using
+Install using:
 
-```
-pip install tox
-```
+.. code-block::
 
-and then, from the `commissaire-http` folder, run the following
+   $ pip install tox
+
+
+and then, from the ``commissaire-http`` folder, run the following
 command:
 
-```
-tox -v -e py35
-```
+.. code-block:: shell
+
+   $ tox -v -e py35
+
+
+Verification
+------------
+
 
+After this the API will be available at ``http://127.0.0.1:8000/``. To
+verify it works, we will use the initial user ``a`` with pass ``a``.
 
-## Verification
-After this the API will be available at `http://127.0.0.1:8000/`. To
-verify it works, we will use the initial user `a` with pass `a`.
+.. code-block::
 
-```
-curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/
-```
+   $ curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/
```

Comment:
This worked, I think?

```
(devel) [~/rhat/commissaire/commissaire-http] [2] 11:12:26  (master)
$ curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/
DEBUG:StorageService:Received message "eebb321b-ad90-41b6-98a4-d2e61ed9aa34" {'id': '0b757a68-2856-42cd-aa6c-8b44f1d63fc7', 'jsonrpc': '2.0', 'method': 'list', 'params': {'model_type_name': 'Clusters'}}
DEBUG:etcd.client:New etcd client created for http://localhost:2379
DEBUG:store:> LIST <Clusters []...>
DEBUG:etcd.client:Issuing read for key /commissaire/clusters/ with args {'recursive': True}
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost
DEBUG:urllib3.connectionpool:http://localhost:2379 "GET /v2/keys/commissaire/clusters/?recursive=true HTTP/1.1" 200 102
WARNING:EtcdStoreHandler:Etcd returned unserializable data. Skipping the record.TypeError: the JSON object must be str, not 'NoneType'
DEBUG:store:< LIST <Clusters []...>
DEBUG:StorageService:Result for "0b757a68-2856-42cd-aa6c-8b44f1d63fc7": "[]"
DEBUG:StorageService:Responding to response-0b757a68-2856-42cd-aa6c-8b44f1d63fc7
DEBUG:StorageService:Message "eebb321b-ad90-41b6-98a4-d2e61ed9aa34" was ackd
[]127.0.0.1 - - [04/Apr/2017 11:12:49] "GET /api/v0/clusters/ HTTP/1.1" 200 2
```

And then re-running it one more time for great good

```
(devel) [~/rhat/commissaire/commissaire-http] [2] 11:12:49  (master)
$ curl -u "a:a" -X GET http://127.0.0.1:8000/api/v0/clusters/
DEBUG:StorageService:Received message "e7fee5cb-60c2-4155-8ab8-d0d62307ba89" {'id': 'ccd39943-3448-4506-973f-1f81d1d979b7', 'jsonrpc': '2.0', 'method': 'list', 'params': {'model_type_name': 'Clusters'}}
DEBUG:store:> LIST <Clusters []...>
DEBUG:etcd.client:Issuing read for key /commissaire/clusters/ with args {'recursive': True}
DEBUG:urllib3.connectionpool:http://localhost:2379 "GET /v2/keys/commissaire/clusters/?recursive=true HTTP/1.1" 200 102
WARNING:EtcdStoreHandler:Etcd returned unserializable data. Skipping the record.TypeError: the JSON object must be str, not 'NoneType'
DEBUG:store:< LIST <Clusters []...>
DEBUG:StorageService:Result for "ccd39943-3448-4506-973f-1f81d1d979b7": "[]"
DEBUG:StorageService:Responding to response-ccd39943-3448-4506-973f-1f81d1d979b7
DEBUG:StorageService:Message "e7fee5cb-60c2-4155-8ab8-d0d62307ba89" was ackd
127.0.0.1 - - [04/Apr/2017 11:12:58] "GET /api/v0/clusters/ HTTP/1.1" 200 2
[]
```

I changed the `debug` parameter in one config file from `false` to `true`. So that probably explains all the background noise there.